pkg/apis/batch/types.go

```diff
@@ -170,6 +170,12 @@ type JobStatus struct {
 	// +optional
 	CompletionTime *metav1.Time
 
+	// Represents time when the job was failed. It is not guaranteed to
```

Comment:
Please do not extend the current API, it's sufficient as is. You should just ensure to fill in `CompletionTime` and the job status will properly reflect that it failed due to deadline.

Reply:
@soltysh thank you for the review, that was actually my first fix, but I added the `FailureTime` field per @janetkuo request:
> @eladm26 thanks for your contribution. Here are some changes we need instead.
> 
>     * Here's an issue about how we can improve API docs: #68712
> 
>     * Make the user guilde more clear, too https://kubernetes.io/docs/concepts/workloads/controllers/jobs-run-to-completion/
> 
>     * Add an API field in Job that records the time when the Job fails

so just to be clear, is that the final decision?